src/FormField/ScopeBuilder.php

```diff
@@ -0,0 +1,595 @@
+<?php
+
+namespace atk4\ui\FormField;
+
+use atk4\data\Model;
+use atk4\ui\Exception;
+use atk4\ui\Template;
+use atk4\data\Field;
+use atk4\data\Model\Scope\Condition;
+use atk4\data\Model\Scope\Scope;
+use atk4\data\Model\Scope\AbstractScope;
+
+class ScopeBuilder extends Generic
+{
+    /**
+     * Field type specific options.
+     *
+     * @var array
+     */
+    public $options = [
+        'enum' => [
+            'limit' => 250
+        ]
+    ];
+    /**
+     * Max depth of nested conditions allowed.
+     * Corresponds to VueQueryBulder maxDepth
+     *
+     * @var integer
+     */
+    public $maxDepth = 10;
```

Comment:
Have this any performance side effect on ui side if large? (test with 10000)

If not, let's set at least to 25.

Reply:
No performance issues but this will be a hell of a Scope 😄 
I have never used more than 3 - 4 sub-levels.
Not sure if you got what this setting means.
You can always set a value when creating the ScopeBuilder object in your code

Reply:
But in rare cases you may need 11, not that far from 3-4 :D